### ISSUE TITLE
[web] add "radiogroup" semantic role for radio button components

### ISSUE THREAD

**Issue by jcollins-disney** [flutter/flutter#162093]
### Use case

All radio button components (`role="radio"`) must be contained in or owned by an element with `role="radiogroup"`. This role is currently missing from the list of available semantics roles in existing widget libraries.

### Proposal

Add the ability to add WAI-ARIA `role="radiogroup"` to radio button container/parent elements.

### Steps

* [x] Add `AccessibilityRole.radioGroup` to `dart:ui`: https://github.com/flutter/flutter/pull/164154
* [x] Update Flutter Web engine: https://github.com/flutter/flutter/pull/164154

See also these follow-ups:

* https://github.com/flutter/flutter/issues/164584
* https://github.com/flutter/flutter/issues/164589

**Comment by loic-sharma** [comment]
I introduced a [`SemanticsRole.radioGroup`](https://main-api.flutter.dev/flutter/dart-ui/SemanticsRole.html#radioGroup) role. Currently, Material and Cupertino widgets do not add this role automatically, apps have to add this role manually. For example:

```dart
import 'dart:ui';

import 'package:flutter/material.dart';
import 'package:flutter/semantics.dart';

void main() {
  runApp(const RadioExampleApp());
  SemanticsBinding.instance.ensureSemantics();
}

class RadioExampleApp extends StatelessWidget {
  const RadioExampleApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(title: const Text('Radio Sample')),
        body: const Center(child: RadioExample()),
      ),
    );
  }
}

class RadioExample extends StatefulWidget {
  const RadioExample({super.key});

  @override
  State<RadioExample> createState() => _RadioExampleState();
}

class _RadioExampleState extends State<RadioExample> {
  int _groupValue = 0;

  @override
  Widget build(BuildContext context) {
    return Semantics(
      label: 'Radio group',
      role: SemanticsRole.radioGroup,
      explicitChildNodes: true,
      child: Column(
        children: <Widget>[
          ListTile(
            title: const Text('Foo'),
            leading: Radio<int>(
              value: 0,
              groupValue: _groupValue,
              onChanged: (int? value) => setState(() => _groupValue = value ?? 0),
            ),
          ),
          ListTile(
            title: const Text('Bar'),
            leading: Radio<int>(
              value: 1,
              groupValue: _groupValue,
              onChanged: (int? value) => setState(() => _groupValue = value ?? 0),
            ),
          ),
        ],
      ),
    );
  }
}
```

**Comment by yjbanov** [comment]
I filed follow-up issues for material and cupertino widgets here:

* https://github.com/flutter/flutter/issues/164584
* https://github.com/flutter/flutter/issues/164589

This is fixed now.
